documentation/style-guide.rst

```diff
@@ -196,6 +199,35 @@ language):
     achieve the same effect.  This assures that files are flushed and file
     descriptor resources are released in a timely manner.
 
+
+Author attribution
+==================
+
+For new documentation, do not use a byline (naming the author of the document).
+Explicit attribution tends to discourage other users from updating community
+documentation.
+
+Existing documentation with bylines will not be changed unless the author
+decides to do so. This is subject to change in the future.
+
+
+First-person language
+=====================
+
+Avoid first-person language.  Our documentation is owned and maintained by a
+community.  First-person language mistakenly implies a single author.
+
+
+Pronunciation of dunders
+========================
+
+"Dunder methods" like ``__init__`` can be awkward in running prose: is it "an
+__init__" or "a __init__"?  A `quick poll`__ indicates we should use
+"an __init__."
```

Comment:
I'd perhaps phrase this according to the underlying rule that this draws out: that the pronunciation-guided spelling is effectively 'an init method' -- you use the first vowel sound of the word to decide a/an and ignore the underscores.

(Obviously we can also recommend the definite article!)

Reply:
Instead of using a question/answer, I would use a statement that says to ignore the underscore and use the article that matches the first sound (like @AA-Turner suggested).  The quick poll could be a footnote.

Reply:
I generally agree with being guided by pronunciation, however `__init__` is often pronounced as "dunder init", leading to "a dunder init method".

This makes the pronunciation rule difficult to apply in this case, hence this specific dunder rule.

Reply:
Thanks, I've expanded on this paragraph.

Reply:
I might be confused here: is the intend to use 'an' for *every* dunder method, or just those that start with a vowel sound? ``init`` is a member of both sets.

Taking an example of a method starting with a consonant sound, would we say "a `__complex__` method" or "an `__complex__` method"? To me, the first sounds right, following the don't-pronounce-underscores rule, but I would argue it also follows the replace-underscores-with-dunder rule.

I missed the original poll as it didn't come up on Discourse, sorry!

A

Reply:
Ignore me, the most recent commit clears this up. Thanks @nedbat!

A